UnityProject/Assets/Scripts/UI/Systems/MainHUD/UI Bottom/UiChangeling.cs

```diff
@@ -0,0 +1,159 @@
+using Mirror;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using Systems.Antagonists;
+using TMPro;
+using UI.Core;
+using UI.Core.Action;
+using UnityEngine;
+
+namespace Changeling
+{
+	public class UiChangeling : MonoBehaviour
+	{
+		public ChangelingMain ChangelingMain;
+		public static UiChangeling instance;
+		[SerializeField] private TMP_Text chemText = null;
+		[SerializeField] private GameObject chems = null;
+		[SerializeField] private GameObject storeGameObject = null;
+		[SerializeField] private UiChangelingStore store = null;
+
+		[SerializeField] private GameObject memoriesGameObject = null;
+		[SerializeField] private UiChangelingMemories memories = null;
+
+		[SerializeField] private TMP_Text abilityPointsCount = null;
+
+		public List<ChangelingData> ChangelingDataToBuy => ChangelingMain.AllAbilities;
+
+		private void Awake()
+		{
+			instance = this;
+			if (ChangelingMain == null)
+				TurnOff();
+		}
+
+		public void OnDisable()
+		{
+			TurnOff();
+		}
+
+		public void SetUp(ChangelingMain changeling)
+		{
+			ChangelingMain = changeling;
+			RefreshUI();
+			chems.SetActive(true);
+		}
+
+		public void ResetAbilites()
+		{
+			ChangelingMain.ResetAbilities();
+		}
+
+		public void RefreshAbilites()
+		{
+			store.Refresh(ChangelingDataToBuy, ChangelingMain);
+		}
+
+		public void UpdateChemText()
+		{
+			chemText.text = ChangelingMain.Chem.ToString();
+		}
+
+		public void UpdateEPText()
+		{
+			abilityPointsCount.text = $"Left genetic points {ChangelingMain.EvolutionPoints}";
+		}
+
+		public void TurnOff()
+		{
+			UpdateManager.Remove(CallbackType.PERIODIC_UPDATE, RefreshUI);
+
+			if (chems != null)
+				chems.SetActive(false);
+			if (storeGameObject != null)
+				storeGameObject.SetActive(false);
+			if (memoriesGameObject != null)
+				memoriesGameObject.SetActive(false);
+			gameObject.SetActive(false);
+		}
+
+		public void AddAbility(ChangelingData abilityToAdd)
+		{
+			ChangelingMain.CmdBuyAbility(abilityToAdd.Index);
+		}
+
+		public void RefreshUI()
+		{
+			try
+			{
+				UpdateChemText();
+				UpdateEPText();
+			}
+			catch (Exception)
```

Comment:
[try chat is never logged, which is the same as leaving it empty. Log the exception to the console using Logger](https://unitystation.github.io/unitystation/contribution-guides/Development-Standards-Guide/#2-performance-gotchas)

Reply:
I think will be better just remove that